core/src/main/scala/cats/derived/contravariant.scala

```diff
@@ -0,0 +1,95 @@
+trait MkContravariant[F[_]] extends Contravariant[F] {
+  def safeContramap[A, B](fa: F[A])(f: B => Eval[A]): Eval[F[B]]
+  def contramap[A, B](fa: F[A])(f: B => A): F[B] = safeContramap(fa)(f andThen Eval.now).value
+}
+object MkContravariant extends MkContravariantDerivation {
+  def apply[F[_]](implicit F: MkContravariant[F]): MkContravariant[F] = F
+}
+private[derived] abstract class MkContravariantDerivation extends MkContravariantNested {
+  implicit val mkContraHNil: MkContravariant[Const[HNil]#λ] = mkContraConst
+  implicit val mkContraCNil: MkContravariant[Const[CNil]#λ] = mkContraConst
+
+  implicit def mkContraConst[T]: MkContravariant[Const[T]#λ] =
+    new MkContravariant[Const[T]#λ] {
+      def safeContramap[A, B](t: T)(f: B => Eval[A]): Eval[T] = Eval.now(t)
+    }
+}
+
+private[derived] abstract class MkContravariantNested extends MkContravariantCons {
+  protected type FunctorOrMk[F[_]] = Functor[F] OrElse MkFunctor[F]
+  protected def mkSafeMap[F[_], A, B](F: FunctorOrMk[F])(fa: F[A])(f: A => Eval[B]): Eval[F[B]] =
+    F.unify match {
+      case mk: MkFunctor[F] => mk.safeMap(fa)(f)
+      case p => Eval.later(p.map(fa)(f(_).value))
+    }
+
+  implicit def mkContravariantFunctorNested[F[_]](implicit F: Split1[F, FunctorOrMk, ContraOrMk]): MkContravariant[F] =
+    new MkContravariant[F] {
+      def safeContramap[A, B](fa: F[A])(f: B => Eval[A]): Eval[F[B]] =
+        mkSafeMap(F.fo)(F.unpack(fa))(
+          (a: F.I[A]) => mkContraSafe(F.fi)(a)(f)
+        ).map(F.pack)
+    }
+}
+
+private[derived] abstract class MkContravariantCons extends MkContravariantGeneric {
+  implicit def mkContraHCons[F[_]](implicit F: IsHCons1[F, ContraOrMk, MkContravariant]): MkContravariant[F] =
+    new MkContravariant[F] {
+      def safeContramap[A, B](fa: F[A])(f: B => Eval[A]): Eval[F[B]] =
+        Eval.now(F.unpack(fa)).flatMap { case (fha, fta) =>
+          for {
+            fhb <- mkContraSafe(F.fh)(fha)(f)
+            ftb <- F.ft.safeContramap(fta)(f)
+          } yield F.pack(fhb, ftb)
+        }
+    }
+
+  implicit def mkContraCCons[F[_]](implicit F: IsCCons1[F, ContraOrMk, MkContravariant]): MkContravariant[F] =
+    new MkContravariant[F] {
+      def safeContramap[A, B](fa: F[A])(f: B => Eval[A]): Eval[F[B]] = F.unpack(fa) match {
+        case Left(fha) => mkContraSafe(F.fh)(fha)(f).map(fhb => F.pack(Left(fhb)))
+        case Right(fta) => F.ft.safeContramap(fta)(f).map(ftb => F.pack(Right(ftb)))
+      }
+    }
+}
+
+private[derived] abstract class MkContravariantGeneric {
+  protected type ContraOrMk[F[_]] = Contravariant[F] OrElse MkContravariant[F]
+
+  protected def mkContraSafe[F[_], A, B](F: ContraOrMk[F])(fa: F[B])(f: A => Eval[B]): Eval[F[A]] =
+    F.unify match {
+      case mk: MkContravariant[F] => mk.safeContramap(fa)(f)
+      case p => Eval.later(p.contramap(fa)(f(_).value))
+    }
+
+  implicit def mkFunctorGeneric[F[_]](implicit F: Generic1[F, MkContravariant]): MkContravariant[F] =
```

Comment:
```suggestion
  implicit def mkContraGeneric[F[_]](implicit F: Generic1[F, MkContravariant]): MkContravariant[F] =
```

Reply:
Still need to apply this renaming

Reply:
Huh, so weird, I viscerally remember doing this, must have git fu-d myself into a snafu, thanks.